lib/Backends/OpenCL/OpenCL.cpp

```diff
@@ -199,7 +199,7 @@ cl_program OpenCLFunction::loadProgramFromDiskCache(std::string cacheDirectory,
   if (!llvm::sys::fs::exists(programPath))
     return nullptr;
 
-  uint64_t binarySize;
+  size_t binarySize;
```

Comment:
`size_t` on MacOS is not the same as uint64_t and thus your fix breaks `llvm::sys::fs::file_size(programPath, binarySize);`, because it expects binarySize to be `uint64_t`.

Reply:
@pjaaskel So, you may need to have two different variables as you need two different types at different places.

Reply:
Of course :/ I pushed a fix. Does it work finally for you? (I do not have a MacOS to test with).

Reply:
Yes, it works now!